front/pages/api/w/[wId]/vaults/[vId]/members.ts

```diff
@@ -1,85 +1,58 @@
-import type { GroupType, UserType, WithAPIErrorResponse } from "@dust-tt/types";
+import type { VaultType, WithAPIErrorResponse } from "@dust-tt/types";
 import { PatchGroupRequestBodySchema } from "@dust-tt/types";
+import assert from "assert";
 import { isLeft } from "fp-ts/lib/Either";
 import * as reporter from "io-ts-reporters";
 import type { NextApiRequest, NextApiResponse } from "next";
 
 import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
 import type { Authenticator } from "@app/lib/auth";
-import { GroupResource } from "@app/lib/resources/group_resource";
 import { UserResource } from "@app/lib/resources/user_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
 import { apiError } from "@app/logger/withlogging";
 
-export type GetGroupResponseBody = {
-  group: GroupType & {
-    members: UserType[];
-  };
-};
-
-export type PatchGroupResponseBody = {
-  group: GroupType;
-};
+export interface PatchVaultMembersResponseBody {
+  vault: VaultType;
+}
 
 async function handler(
   req: NextApiRequest,
-  res: NextApiResponse<
-    WithAPIErrorResponse<GetGroupResponseBody | PatchGroupResponseBody>
-  >,
+  res: NextApiResponse<WithAPIErrorResponse<PatchVaultMembersResponseBody>>,
   auth: Authenticator
 ): Promise<void> {
-  const { gId } = req.query;
-
-  if (typeof gId !== "string") {
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
+  if (!vault || !vault.canWrite(auth)) {
```

Comment:
PATCH was limited to admins, now aren't we also allowing vaults members to edit vaults's members ?

Reply:
see `canAdministrate`.